src/api/AppApi.test.js

```diff
@@ -176,3 +197,394 @@ function testInstallApp() {
+    testApiShouldSendCorrectDeleteRequest(fnToTest, validParams, axiosParams);
+  });
+}
+
+function testAddAppTypeToApp() {
+
+  describe('addAppTypeToApp()', () => {
+
+    const fnToTest = AppApi.addAppTypeToApp;
+
+    const validParams = [MOCK_APP_DM.id, MOCK_APP_TYPE_DM.id];
+    const invalidParams = [INVALID_PARAMS_SS, INVALID_PARAMS_SS];
+    const axiosParams = [`/${UPDATE_PATH}/${MOCK_APP_DM.id}/${MOCK_APP_TYPE_DM.id}`];
+
+    testApiShouldReturnPromise(fnToTest, validParams);
+    testApiShouldUseCorrectAxiosInstance(fnToTest, validParams, APP_API);
+    testApiShouldNotThrowOnInvalidParameters(fnToTest, validParams, invalidParams);
+    testApiShouldRejectOnInvalidParameters(fnToTest, validParams, invalidParams);
+    testApiShouldSendCorrectGetRequest(fnToTest, validParams, axiosParams);
+  });
+}
+
+function testRemoveAppTypeFromApp() {
+
+  describe('removeAppTypeFromApp()', () => {
+
+    const fnToTest = AppApi.removeAppTypeFromApp;
+
+    const validParams = [MOCK_APP_DM.id, MOCK_APP_TYPE_DM.id];
+    const invalidParams = [INVALID_PARAMS_SS, INVALID_PARAMS_SS];
+    const axiosParams = [`/${UPDATE_PATH}/${MOCK_APP_DM.id}/${MOCK_APP_TYPE_DM.id}`];
+
+    testApiShouldReturnPromise(fnToTest, validParams);
+    testApiShouldUseCorrectAxiosInstance(fnToTest, validParams, APP_API);
+    testApiShouldNotThrowOnInvalidParameters(fnToTest, validParams, invalidParams);
+    testApiShouldRejectOnInvalidParameters(fnToTest, validParams, invalidParams);
+    testApiShouldSendCorrectDeleteRequest(fnToTest, validParams, axiosParams);
+  });
+}
+
+function testUpdateAppEntitySetConfig() {
+
+  describe('updateAppEntitySetConfig()', () => {
+
+    const fnToTest = AppApi.updateAppEntitySetConfig;
+
+    const validParams = [MOCK_ORGANIZATION_DM.id, MOCK_APP_DM.id, MOCK_APP_TYPE_DM.id, MOCK_ENTITY_SET_DM.id];
+    const invalidParams = [INVALID_PARAMS_SS, INVALID_PARAMS_SS, INVALID_PARAMS_SS, INVALID_PARAMS_SS];
+    const axiosParams = [
+      `/${UPDATE_PATH}/${MOCK_ORGANIZATION_DM.id}/${MOCK_APP_DM.id}/${MOCK_APP_TYPE_DM.id}/${MOCK_ENTITY_SET_DM.id}`];
+
+    testApiShouldReturnPromise(fnToTest, validParams);
+    testApiShouldUseCorrectAxiosInstance(fnToTest, validParams, APP_API);
+    testApiShouldNotThrowOnInvalidParameters(fnToTest, validParams, invalidParams);
+    testApiShouldRejectOnInvalidParameters(fnToTest, validParams, invalidParams);
+    testApiShouldSendCorrectGetRequest(fnToTest, validParams, axiosParams);
+  });
+}
+
+function testUpdateAppConfigPermissions() {
+
+  describe('updateAppConfigPermissions()', () => {
+
+    const fnToTest = AppApi.updateAppConfigPermissions;
+    const permissions = [PermissionTypes.READ, PermissionTypes.WRITE];
+
+    const validParams = [MOCK_ORGANIZATION_DM.id, MOCK_APP_DM.id, MOCK_APP_TYPE_DM.id, permissions];
+    const invalidParams = [
+      INVALID_PARAMS_SS, INVALID_PARAMS_SS, INVALID_PARAMS_SS, [INVALID_PARAMS_SS, INVALID_PARAMS_SS]];
```

Comment:
After 3 params, if they don't fit on one line, each param should be on its own line. Closing bracket should be on its own line.